Assets/Scripts/NetworkClient.cs

```diff
@@ -113,10 +137,17 @@ private IEnumerator TryConnectToServers()
         }
     }
 
-    private async Task ConnectWebSocket(WebSocket websocket, string url)
+    private async Task ConnectWebSocket(WebSocket websocket, string url, FlagObject doAbort)
     {
         websocket.OnOpen += () =>
         {
+            if (doAbort.IsSet())
```

Comment:
And just sanity-checking my understanding: there is absolutely no multi-threading concern here. Async functions in Unity script get run later by Unity on the main Unity thread.

Reply:
Correct.